ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -291,35 +293,111 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     };
   }
 
+  private void primaryServerStartTransaction(
+      final StreamInfo info, final DataStreamRequestByteBuf request, final ChannelHandlerContext ctx) {
+    info.getStream().thenApplyAsync(stream -> {
+      try {
+        RaftClientReply reply = server.submitClientRequest(stream.getRaftClientRequest());
```

Comment:
Please use submitClientRequestAsync.  We should use the async function whenever possible.